### ISSUE TITLE
Can not add the same measure twice on [moduleKey=...]

### ISSUE THREAD

**Issue by rducom** [SonarSource/sonar-csharp#838]
Hello

We have the folowing intermittent issue on our CI pipeline (4 to 6% of builds).

`ERROR: Error during SonarQube Scanner execution
java.lang.UnsupportedOperationException: Can not add the same measure twice on [moduleKey=ilucca:ilucca:28FC956D-ED61-4A74-9C97-947AEC170423, relative=Areas/admin/Angular/Import/app/Import2.aspx.cs, basedir=D:\workspace\masterd05bfe2a\iLucca]: DefaultMeasure[component=[moduleKey=ilucca:ilucca:28FC956D-ED61-4A74-9C97-947AEC170423, relative=Areas/admin/Angular/Import/app/Import2.aspx.cs, basedir=D:\workspace\masterd05bfe2a\iLucca],metric=Metric[id=<null>,key=classes,description=Classes,type=INT,direction=-1,domain=Size,name=Classes,qualitative=false,userManaged=false,enabled=true,worstValue=<null>,bestValue=<null>,optimizedBestValue=false,hidden=false,deleteHistoricalData=false,decimalScale=<null>],value=1,fromCore=false]
	at org.sonar.scanner.sensor.DefaultSensorStorage.saveMeasure(DefaultSensorStorage.java:265)
	at org.sonar.scanner.sensor.DefaultSensorStorage.store(DefaultSensorStorage.java:210)
	at org.sonar.api.batch.sensor.measure.internal.DefaultMeasure.doSave(DefaultMeasure.java:95)
	at org.sonar.api.batch.sensor.internal.DefaultStorable.save(DefaultStorable.java:45)
	at org.sonarsource.dotnet.shared.plugins.protobuf.MetricsImporter.saveMetric(MetricsImporter.java:96)
	at org.sonarsource.dotnet.shared.plugins.protobuf.MetricsImporter.consumeFor(MetricsImporter.java:54)
	at org.sonarsource.dotnet.shared.plugins.protobuf.MetricsImporter.consumeFor(MetricsImporter.java:37)
	at org.sonarsource.dotnet.shared.plugins.protobuf.ProtobufImporter.consume(ProtobufImporter.java:50)
	at org.sonarsource.dotnet.shared.plugins.protobuf.RawProtobufImporter.accept(RawProtobufImporter.java:44)
	at org.sonarsource.dotnet.shared.plugins.AbstractSensor.parseProtobuf(AbstractSensor.java:88)
	at org.sonarsource.dotnet.shared.plugins.AbstractSensor.importResults(AbstractSensor.java:76)
	at org.sonar.plugins.csharp.CSharpSensor.executeInternal(CSharpSensor.java:98)
	at org.sonar.plugins.csharp.CSharpSensor.execute(CSharpSensor.java:69)
	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:88)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:82)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:68)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:179)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:129)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:261)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:256)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:254)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:245)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:129)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:129)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:119)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:116)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:123)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:77)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)`

We use sonar-scanner-msbuild-3.0.0.629 with the sonar-charp plugin in last version (on SonarQube 6.5, but we have the same on 6.3).
Our CI pipeline is runned by Jenkins2, with windows slaves.

Unfortunaltly, we don't have repro steps, if we launch the build again on the same branch or PR, the error usually don't happen again. 

Is there anything to do to prevent this issue ?

Thanks in advance

**Comment by Evangelink** [comment]
Hi @rducom,

Could you provide the full log (preferably with the debug switch)?

Thanks

**Comment by rducom** [comment]
@Evangelink here is the log : https://gist.github.com/rducom/59e9de3eff4ef98d64d68d0965e7aa7b
I can't switch to debug mode since it's our production CI pipeline, and our slaves are already near a 100% usage, I can't switch to debug waiting for the the next fail...

**Comment by Evangelink** [comment]
Thx for the log. I am still not sure where is the bug...Could you provide the `metrics.pb` file located in `.sonarqube\out\**PROJECT_CONTAINING_Import2.aspx.cs**`?

**Comment by rducom** [comment]
Here it is : http://dl.free.fr/fy8NeVHcF
Sorry for the delay, we format the jenkins salve disks each night, I have had to find another failing build.

**Comment by kharaone** [comment]
I have exactly the same error for a long time already but the only references I've found for this error were about Java plugins interferring with each other.
It even prevents me from completely integrating sonarqube to my CI pipeline as the build becomes too brittle.

Cleaning the workspace manually usually allows me to pass the next build but that's definitely not a viable option for my pipeline (Moreover, it is already set up to clean the workspace at the start and end of the build so I don't really understand what's triggering this error)

**Comment by kharaone** [comment]
Any ways we could help debug this ? I'd really like to leverage Sonarqube for my CI and that bug prevents me to integrate it.

**Comment by Evangelink** [comment]
Hi @kharaone, we are currently doing some big changes on the scanner for msbuild side and we hope it will solve this problem. RC shall be available soon (end of the week or beginning of next week).

**Comment by rducom** [comment]
Unfortunalty, today, this issue decided to block all our CI Pipeline.
The only thing changed since last week is I updated sonar to 6.7 LTS

```
INFO: -------------  Scan Lucca.Application
INFO: Base dir: D:\workspace\masterd05bfe2a\Application\Lucca.Application
INFO: Working dir: D:\workspace\masterd05bfe2a\.sonarqube\out\.sonar\mod21
INFO: Source paths: Context/HttpCustomerContext.cs, Context/HttpExecutionContext.cs, Context/HttpServerContext.cs, Properties/AssemblyInfo.cs, app.config, packages.config
INFO: Source encoding: UTF-8, default locale: en_US
WARNING: WARN: Property 'sonar.php.file.suffixes' is not declared as multi-values/property set but was read using 'getStringArray' method. The SonarQube plugin declaring this property should be updated.
INFO: Language is forced to cs
INFO: Index files
INFO: Excluded sources: 
INFO:   **/Static/**/*
INFO:   **/SPA/**/*
INFO: Analyzer working directory contains 6 .pb file(s)
WARNING: WARN: File 'D:\workspace\masterd05bfe2a\Application\Lucca.Application\app.config' is ignored because it doesn't belong to the forced language 'cs'
WARNING: WARN: File 'D:\workspace\masterd05bfe2a\Application\Lucca.Application\packages.config' is ignored because it doesn't belong to the forced language 'cs'
INFO: 4 files indexed
INFO: 0 files ignored because of inclusion/exclusion patterns
INFO: Quality profile for cs: LuccaC#
INFO: Sensor Embedded CSS Analyzer Sensor [css]
INFO: 0 source files to be analyzed
INFO: Sensor Embedded CSS Analyzer Sensor [css] (done) | time=1ms
INFO: Sensor C# [csharp]
INFO: 0/0 source files have been analyzed
INFO: Importing analysis results from D:\workspace\masterd05bfe2a\.sonarqube\out\Lucca.Application_1970\output-cs
INFO: Importing Roslyn report
INFO: Sensor C# [csharp] (done) | time=83ms
INFO: Sensor SonarJavaXmlFileSensor [java]
INFO: Sensor SonarJavaXmlFileSensor [java] (done) | time=1ms
INFO: Sensor Analyzer for "php.ini" files [php]
INFO: Sensor Analyzer for "php.ini" files [php] (done) | time=0ms
```

```
WARNING: WARN: File 'D:\workspace\masterd05bfe2a\Web\Start.Web.Sdk\obj\Release\Start.Web.Sdk.pdb' is ignored because it doesn't belong to the forced language 'cs'
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 1:58.279s
INFO: Final Memory: 53M/291M
INFO: ------------------------------------------------------------------------
ERROR: Error during SonarQube Scanner execution
ERROR: File Application/Lucca.Application/Context/HttpCustomerContext.cs can't be indexed twice. Please check that inclusion/exclusion patterns produce disjoint sets for main and test files
```

Here's the metrics.pb : http://dl.free.fr/hGVPB1ypK

Is there a workaround ? (other than disable sonar completly?)

**Comment by rducom** [comment]
@Evangelink have you published a preview version of msbuild-scanner ?

**Comment by rducom** [comment]
So bad news : I rolled back to Sonar 6.5, and this blocking issue doesn't happens anymore....
This issue was systematic on sonar 6.7. It's only sporadic on 6.5.
Hope this can help ^^

**Comment by Evangelink** [comment]
Hi @rducom,
We have send the RFF for the scanner for msbuild 4.0 and SonarC# 6.7. We have another related bug (see #958 which seems to be caused when you have multiple csproj at the same level. Is it the case on your side?

**Comment by rducom** [comment]
No, all our csproj are in separate folders. However, we have multiple SLN in the same folder, but only one in being built at the same time.
I tried to exclude mentioned files with scanner argument without luck : Assemblyinfo.cs was the next "duplicated file". I gave up... and rolled back.

**Comment by bellingard** [comment]
@henryju Could it be related to the changes we did in 6.6?

**Comment by rducom** [comment]
Tried again with 6.7LTS, with sonar-scanner-msbuild-3.0.0.629 & sonar-scanner-msbuild-4.0.0.731, and with and without sonar-csharp 6.7-RC1. This time, I can't get pas the sonar start step.
I mailed the detailled exception with the RFF for sonar-csharp 6.7 mail.

```
Unhandled Exception: System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at SonarQube.TeamBuild.PreProcessor.SonarWebService.<>c__DisplayClass8_2.<GetActiveRules>b__1(JObject r)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at SonarQube.TeamBuild.PreProcessor.SonarWebService.GetActiveRules(String qprofile)
   at SonarQube.TeamBuild.PreProcessor.TeamBuildPreProcessor.FetchArgumentsAndRulesets(ISonarQubeServer server, ProcessedArgs args, TeamBuildSettings settings, IDictionary`2& serverSettings, List`1& analyzersSettings)
   at SonarQube.TeamBuild.PreProcessor.TeamBuildPreProcessor.DoExecute(ProcessedArgs localSettings)
   at SonarQube.TeamBuild.PreProcessor.TeamBuildPreProcessor.Execute(String[] args)
   at SonarQube.Bootstrapper.BootstrapperClass.PreProcess()
   at SonarQube.Bootstrapper.BootstrapperClass.Execute()
   at SonarQube.Bootstrapper.Program.Execute(String[] args, ILogger logger)
   at SonarQube.Bootstrapper.Program.Main(String[] args)
```
Exception happen in both commit and preview mode.
I have made the same migration steps as yesterday (when I was reproducing the "duplicate key twice" issue). I also have double checked the new quality profiles were selected.... 
No luck at all...
Very disappointing...

**Comment by henryju** [comment]
Many different problems, so this thread starts to be hard to follow.
* duplicate measure
* file can't be indexed twice -> need full log of this analysis
* Value cannot be null -> here I suspect a corruption of the Elastic Search index. Try dropping the data/es folder on the server, and restart it.

**Comment by dbmeneses** [comment]
Is it possible that you have parallel analysis running on the same location?
If the problem with file indexed twice happens on the same project as the one you provide full logs, it doesn't seem to be related to #958 as the project structure seems to be OK. The fact that it's intermittent suggests it could be a problem of concurrent builds.

**Comment by rducom** [comment]
### Can not add the same measure twice on [moduleKey=...] :

On each of our Jenkins slaves, we have 4 build slots, so yes, there are times when different slots are building in //.
However : 
- each slot is used by only one Jenkins Job at a time
- each job in executed in a separated folder
- since only one job is running at a time, each sonar analysis is using same Key, but distinct `/d:"sonar.github.pullRequest` id

Since current CPU server have plenty of CPU core available, I would be astonished if you tell me I can't run multiple distinct analysis in // on the same server ^^. 
Tell me if this is the case, because I've never seen such requirement on sonar-runner and I'm using it since years.

### File can't be indexed twice  :
sorry, no log available, I've rolled back to 6.5 since this issue was totally blocking
Since the message is really near the first one, I was believing it was the same issue. 
Can you tell me precisely which log do you need ?

**Comment by henryju** [comment]
For "file can't be indexed twice" I would need the configuration of sonar.sources/sonar.tests of each module, as well as module base directory
Something like:

> INFO: Base dir: D:\workspace\masterd05bfe2a\Application\Lucca.Application
> INFO: Source paths: Context/HttpCustomerContext.cs, Context/HttpExecutionContext.cs, Context/HttpServerContext.cs, Properties/AssemblyInfo.cs, app.config, packages.config

for each module.

**Comment by Evangelink** [comment]
@rducom Are you still experiencing the issue? If you do, are you still using the same versions of SonarC#, Scanner for MSBuild and SonarQube?

**Comment by rducom** [comment]
Hi @Evangelink , just tried to update Sonar from 6.5 to 6.7.2 LTS.
Now the issue is systematic...
Parrallel builds are not a root cause, this issue happen now while builds are alone on each slave.

`ERROR: File Application/Lucca.Application/Context/HttpContextHelper.cs can't be indexed twice. Please check that inclusion/exclusion patterns produce disjoint sets for main and test files`

@henryju I can give you a full log, with the sonar folder (.sonarqube) by private mail.
For what I see, there's only one time where the mentionned file appears in `Source paths`

```
[backend] INFO: Base dir: D:\workspace\PR-6794b20dc967\Application\Lucca.Application
[backend] INFO: Working dir: D:\workspace\PR-6794b20dc967\.sonarqube\out\.sonar\ilucca_ilucca_D7963650-2ED7-4817-AAF3-4EBFC434A3FD
[backend] INFO: Source paths: Context/HttpContextHelper.cs, Context/HttpCustomerContext.cs, Context/HttpExecutionContext.cs, Context/HttpServerContext.cs, Properties/AssemblyInfo.cs
[backend] INFO: Source encoding: UTF-8, default locale: en_US
[backend] INFO: Language is forced to cs
[backend] INFO: Index files
[backend] INFO: Excluded sources: 
[backend] INFO:   **/Static/**/*
[backend] INFO:   **/SPA/**/*
[backend] INFO:   **/bin/**/*
[backend] INFO: 5 files indexed
```

The only other place where HttpContextHelper.cs appears in log, is where the error is mentionned.

For now I'll rolllback to 6.5, were the issue is only sporadic

**Comment by Evangelink** [comment]
Hi @rducom,

Can you send me the log at amaury DOT leve AT sonarsource DOT com

**Comment by rducom** [comment]
email sent. thank you !

**Comment by valhristov** [comment]
Hi @rducom, I managed to reproduce the problem locally by setting the `/d:sonar.sources` property. I suspect you are setting it too, because your `.sonarqube\out\sonar-project.properties` file contains it twice. Please, remove the property from the Scanner for MSBuild command line and the problem should disappear.

This problem should appear consistently when using SonarC# 6.0+ and SonarQube 6.4+ (as far as I can remember SonarQube included SonarC# 5.11 by default until version 6.7, hence if you use the default installation of SonarQube 6.5 the problem does not appear).

I need to investigate more to see what's the expected behavior when this property is set. It seems that the scanner does not need it as command line argument.

**Comment by rducom** [comment]
Ok, @valhristov I'll give a try on 6.5 and attempt another migration to 6.7.2 asap.
Really thank you for your effort !

**Comment by rducom** [comment]
I have a really good news : it works on 6.7.2 !!! 

One issue however : on the first run, I have had the following exception : 

```
Unhandled Exception: System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at SonarQube.TeamBuild.PreProcessor.SonarWebService.<>c__DisplayClass8_2.<GetActiveRules>b__1(JObject r)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at SonarQube.TeamBuild.PreProcessor.SonarWebService.GetActiveRules(String qprofile)
   at SonarQube.TeamBuild.PreProcessor.TeamBuildPreProcessor.FetchArgumentsAndRulesets(ISonarQubeServer server, ProcessedArgs args, TeamBuildSettings settings, IDictionary`2& serverSettings, List`1& analyzersSettings)
   at SonarQube.TeamBuild.PreProcessor.TeamBuildPreProcessor.DoExecute(ProcessedArgs localSettings)
   at SonarQube.TeamBuild.PreProcessor.TeamBuildPreProcessor.Execute(String[] args)
   at SonarQube.Bootstrapper.BootstrapperClass.PreProcess()
   at SonarQube.Bootstrapper.BootstrapperClass.Execute()
   at SonarQube.Bootstrapper.Program.Execute(String[] args, ILogger logger)
   at SonarQube.Bootstrapper.Program.Main(String[] args)
```

which is weird, because I just remove the "sources" parameter, so of course it's null.... And in fact, the command line "sources" parameter have nothing to do with the exception "source" parameter 😄 

I have had to switch to the new default quality profile in order to make the runner works. (thanks to google groups archive)

Hope this "sources" parameter will be deprecated, or at least better documented !

Many thanks again @valhristov and @Evangelink, this issue have been preventing us to upgrade since 6 months, and have been failing a lot of builds !

**Comment by valhristov** [comment]
@rducom, this exception happens when there are no active rules in the quality profile. It looks like something we need to fix as soon as possible, I created a ticket: #455

I am closing this ticket now, please, let us know if you have additional problems.

**Comment by kiransterling** [comment]
I am facing the same issue . Let me know what is the resolution.

**Comment by bmorenc** [comment]
We are using SonarQube 6.7.7 with SonarC# 7.9.1 and this issue is still present.  Why was this closed?  We use the coverity-sonar-plugin and it catches exceptions while attempting to add metrics, for example, "ncloc".  However if SonarC# runs after this, SonarC# throws the UnsupportedOperationException and breaks the whole job.  

There should (I think) be some exception handling here to catch any duplicate metrics added by other plugins:

[MetricsImporter.java saveMetric](https://github.com/SonarSource/sonar-dotnet/blob/master/sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/MetricsImporter.java#L81)

```
18:57:47.198 ERROR: Error during SonarQube Scanner execution
 java.lang.UnsupportedOperationException: Can not add the same measure twice on <filename>: DefaultMeasure[component=<filename>,metric=Metric[id=<null>,key=ncloc,description=Non commenting lines of code,type=INT,direction=-1,domain=Size,name=Lines of Code,qualitative=false,userManaged=false,enabled=true,worstValue=<null>,bestValue=<null>,optimizedBestValue=false,hidden=false,deleteHistoricalData=false,decimalScale=<null>],value=103,fromCore=false]
 	at org.sonar.scanner.sensor.DefaultSensorStorage.saveMeasure(DefaultSensorStorage.java:268)
 	at org.sonar.scanner.sensor.DefaultSensorStorage.store(DefaultSensorStorage.java:212)
 	at org.sonar.api.batch.sensor.measure.internal.DefaultMeasure.doSave(DefaultMeasure.java:95)
 	at org.sonar.api.batch.sensor.internal.DefaultStorable.save(DefaultStorable.java:45)
 	at org.sonarsource.dotnet.shared.plugins.protobuf.MetricsImporter.saveMetric(MetricsImporter.java:89)
 	at org.sonarsource.dotnet.shared.plugins.protobuf.MetricsImporter.consumeFor(MetricsImporter.java:71)
 	at org.sonarsource.dotnet.shared.plugins.protobuf.MetricsImporter.consumeFor(MetricsImporter.java:35)
 	at org.sonarsource.dotnet.shared.plugins.protobuf.ProtobufImporter.consume(ProtobufImporter.java:69)
 	at org.sonarsource.dotnet.shared.plugins.protobuf.RawProtobufImporter.accept(RawProtobufImporter.java:44)
 	at org.sonarsource.dotnet.shared.plugins.ProtobufDataImporter.parseProtobuf(ProtobufDataImporter.java:103)
 	at org.sonarsource.dotnet.shared.plugins.ProtobufDataImporter.importResults(ProtobufDataImporter.java:71)
 	at org.sonar.plugins.csharp.CSharpSensor.executeInternal(CSharpSensor.java:91)
 	at org.sonar.plugins.csharp.CSharpSensor.execute(CSharpSensor.java:69)
 	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
 	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:88)
 	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:82)
 	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:73)
 	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:88)
 	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:177)
 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
 	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:291)
 	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:286)
 	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:264)
 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
 	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
 	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
 	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
 	at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
 	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:111)
 	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
 	at java.lang.reflect.Method.invoke(Unknown Source)
 	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
 	at com.sun.proxy.$Proxy0.execute(Unknown Source)
 	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
 	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
 	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:123)
 	at org.sonarsource.scanner.cli.Main.execute(Main.java:77)
 	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
```

Thank you,
Bryan

**Comment by pavel-mikula-sonarsource** [comment]
Hi @bmorenc,

The issue was closed, because it was [fixed in 6.7.2](https://github.com/SonarSource/sonar-dotnet/issues/838#issuecomment-374393188) according to comment above. 

Seeing it on 6.7.7 is unexpected, but I'm afraid we cannot do anything about it on that very old version. Please create a new issue with relevant details if you can reproduce it with the latest version.

**Comment by bmorenc** [comment]
@pavel-mikula-sonarsource thank you for the reply, this issue was 3 issues in one so it was a bit touch to follow the resolution to each specific piece.  I can start a new issue.  Indeed we are sitting on a very old version, will a fix into SonarC# 8.x even be compatible with 6.7.7?  Sadly I am not in control of the SQ installation so I'll have to pursue our responsible team for why they are sticking with a version so far out of support.

**Comment by andrei-epure-sonarsource** [comment]
@bmorenc we dropped support for SQ 6.7 in v8.0, so the latest version with SQ 6.7 support would be 7.17
